File: GoChat/src/main/java/com/jw/gochat/utils/BackTaskFactory.java
package com.jw.gochat.utils;

import com.jw.gochat.bean.NetTask;

import java.util.HashMap;

public class BackTaskFactory {

	public static NetTask newFriendAcceptTask(String invitor, String acceptor) {

		NetTask task = new NetTask();
		task.setMethod("POST");

		HashMap<String, String> params = new HashMap<String, String>();
		params.put("invitor", invitor);
		params.put("acceptor", acceptor);

		task.setParams(params);
		task.setPath("/friend/accept");
		task.setProtocol("HTTP");
		return task;
	}

	public static NetTask userIconChangeTask(String name,String iconPath) {
		NetTask task = new NetTask();
		task.setMethod("POST");
		task.setType(NetTask.TYPE_UPLOAD);
		HashMap<String, String> files = new HashMap<String, String>();
		files.put(name+".png", iconPath);
		HashMap<String, String> params = new HashMap<String, String>();
		task.setParams(params);
		task.setFiles(files);
		task.setPath("/user/icon");
		task.setProtocol("HTTP");
		return task;
	}

	public static NetTask userNameChangeTask(String name) {
		NetTask task = new NetTask();
		task.setMethod("POST");
		HashMap<String, String> params = new HashMap<String, String>();
		params.put("name", name);
		task.setParams(params);
		task.setPath("/user/name");
		task.setProtocol("HTTP");
		return task;
	}
}

File: goChat/src/main/java/com/jw/gochat/utils/BackTaskFactory.kt
package com.jw.gochat.utils

import com.jw.business.model.bean.NetTask
import java.util.HashMap

object BackTaskFactory {

    fun newFriendAcceptTask(invitor: String, acceptor: String): NetTask {
        val task = NetTask()
        task.method = "POST"
        val params = HashMap<String, String>()
        params["invitor"] = invitor
        params["acceptor"] = acceptor
        task.params = params
        task.path = "/friend/accept"
        task.protocol = "HTTP"
        return task
    }

    fun userIconChangeTask(name: String, iconPath: String): NetTask {
        val task = NetTask()
        task.method = "POST"
        task.type = NetTask.TYPE_UPLOAD
        val files = HashMap<String, String>()
        files["$name.png"] = iconPath
        val params = HashMap<String, String>()
        task.params = params
        task.files = files
        task.path = "/user/invitator_icon"
        task.protocol = "HTTP"
        return task
    }

    fun userNameChangeTask(name: String): NetTask {
        val task = NetTask()
        task.method = "POST"
        val params = HashMap<String, String>()
        params["invitator_name"] = name
        task.params = params
        task.path = "/user/invitator_name"
        task.protocol = "HTTP"
        return task
    }
}